symengine/derivative.cpp

```diff
@@ -49,6 +50,8 @@ static RCP<const Basic> diff(const CLASS &self, \
     DIFF0(LeviCivita)
     DIFF0(Max)
     DIFF0(Min)
+    DIFF0(Interval)
+    DIFF0(EmptySet)
```

Comment:
Throwing an error is better. It doesn't make sense for an interval to have a derivative